includes/validation/class-amp-validation-error-taxonomy.php

```diff
@@ -1196,38 +1398,62 @@ public static function filter_terms_clauses_for_description_search( $clauses, $t
+		if ( ! ( self::TAXONOMY_SLUG === get_current_screen()->taxonomy || AMP_Invalid_URL_Post_Type::POST_TYPE_SLUG === get_current_screen()->post_type ) ) { // WPCS: CSRF ok.
 			return;
 		}
-		$actioned = sanitize_key( $_GET['amp_actioned'] ); // WPCS: CSRF ok.
-		$count    = intval( $_GET['amp_actioned_count'] ); // WPCS: CSRF ok.
-		$message  = null;
-		if ( self::VALIDATION_ERROR_ACCEPT_ACTION === $actioned ) {
-			$message = sprintf(
-				/* translators: %s is number of errors accepted */
-				_n(
-					'Accepted %s error. It will no longer block related URLs from being served as AMP.',
-					'Accepted %s errors. They will no longer block related URLs from being served as AMP.',
-					number_format_i18n( $count ),
-					'amp'
-				),
-				$count
-			);
-		} elseif ( self::VALIDATION_ERROR_REJECT_ACTION === $actioned ) {
-			$message = sprintf(
-				/* translators: %s is number of errors rejected */
-				_n(
-					'Rejected %s error. It will continue to block related URLs from being served as AMP.',
-					'Rejected %s errors. They will continue to block related URLs from being served as AMP.',
-					number_format_i18n( $count ),
-					'amp'
-				),
-				$count
-			);
+
+		// Show success messages for accepting/rejecting validation errors.
+		if ( ! empty( $_GET['amp_actioned'] ) && ! empty( $_GET['amp_actioned_count'] ) ) {
+			$actioned = sanitize_key( $_GET['amp_actioned'] ); // WPCS: CSRF ok.
+			$count    = intval( $_GET['amp_actioned_count'] ); // WPCS: CSRF ok.
+			$message  = null;
+			if ( self::VALIDATION_ERROR_ACCEPT_ACTION === $actioned ) {
+				$message = sprintf(
+					/* translators: %s is number of errors accepted */
+					_n(
+						'Accepted %s error. It will no longer block related URLs from being served as AMP.',
+						'Accepted %s errors. They will no longer block related URLs from being served as AMP.',
+						number_format_i18n( $count ),
+						'amp'
+					),
+					$count
+				);
+			} elseif ( self::VALIDATION_ERROR_REJECT_ACTION === $actioned ) {
+				$message = sprintf(
+					/* translators: %s is number of errors rejected */
+					_n(
+						'Rejected %s error. It will continue to block related URLs from being served as AMP.',
+						'Rejected %s errors. They will continue to block related URLs from being served as AMP.',
+						number_format_i18n( $count ),
+						'amp'
+					),
+					$count
+				);
+			}
+
+			if ( $message ) {
+				printf( '<div class="notice notice-success is-dismissible"><p>%s</p></div>', esc_html( $message ) );
+			}
 		}
 
-		if ( $message ) {
-			printf( '<div class="notice notice-success is-dismissible"><p>%s</p></div>', esc_html( $message ) );
+		// Show success message for clearing empty terms.
+		if ( isset( $_GET[ self::VALIDATION_ERRORS_CLEARED_QUERY_VAR ] ) ) { // WPCS: csrf ok.
```

Comment:
This looks good. After 'forgetting' some errors and clicking 'Clear empty,' this appeared:

<img width="1216" alt="after-clear-empty" src="https://user-images.githubusercontent.com/4063887/45891653-7f1db000-bd9c-11e8-85d0-0ebac9d2bedf.png">